### PR TITLE
Feature/Admin Orders

### PR DESCRIPTION
**Pull request by rishabh-vinsol**

### CODE REVIEW

File: db/migrate/20230608133616_add_url_slug_to_branch.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddUrlSlugToBranch < ActiveRecord::Migration[7.0]
+  def change
+    add_column :branches, :url_slug, :string
```
Comment (Sparsh-Goyal-2000) [inline db/migrate/20230608133616_add_url_slug_to_branch.rb:3]: Add a script to update old branches data

File: config/routes.rb [hunk 2]
```diff
@@ -37,6 +37,9 @@
         post 'add_meal', to: 'branches#create_meal'
         get :toggle_meal_active
         get :toggle_meal_inactive
+        post :order_ready
+        post :order_picked_up
+        post :order_cancelled
```
Comment (Sparsh-Goyal-2000) [inline config/routes.rb:42]: i think we should add these routes on orders as we already have branch id there.
Just authorize admin before these actions.

Comment (Sparsh-Goyal-2000) [inline config/routes.rb:42]: then you can rename them as mark_`state_name`

File: app/controllers/admin/branches_controller.rb [hunk 3]
```diff
@@ -32,12 +37,11 @@ def update
...
   def destroy
     @branch.destroy ? flash[:notice] = t('.success') : flash[:alert] = t('.failure')
-
-    redirect_to branches_url
+    redirect_to branches_path
   end
 
   def meals
-    @branch_meals = @branch.branches_meals.includes(:meal, :branch).order(:id)
+    @branch_meals = @branch.branches_meals.includes(:meal, :branch).order(:placed_on)
```
Comment (Sparsh-Goyal-2000) [inline app/controllers/admin/branches_controller.rb:44]: `placed_on ` does not exists on branch meals.

File: app/controllers/orders_controller.rb [hunk 4]
```diff
@@ -68,6 +68,7 @@ def cancel_order
     if @order.cancelled
       flash[:notice] = 'Order was cancelled successfully'
     else
+      @order.update_inventory(false)
```
Comment (Sparsh-Goyal-2000) [inline app/controllers/orders_controller.rb:71]: merge main
then we will not require this change

File: app/models/order.rb [hunk 5]
```diff
@@ -42,6 +42,14 @@ def received
     update(status: :received, placed_on: Time.now)
   end
 
+  def ready
```
Comment (Sparsh-Goyal-2000) [inline app/models/order.rb:45]: make_ready

File: app/models/order.rb [hunk 6]
```diff
@@ -42,6 +42,14 @@ def received
...
+    update(status: :ready)
+  end
+
+  def picked_up
```
Comment (Sparsh-Goyal-2000) [inline app/models/order.rb:49]: mark_picked_up

File: app/views/admin/branches/show.html.erb [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  <div class="border p-4">
+    <div class="d-flex justify-content-between align-items-center mb-4">
+      <h2><%= link_to "Branches", branches_path, class: "text-decoration-none" %> / <%= @branch.name %> / Orders</h2>
+    </div>
+    <% if @orders.blank? %>
+      <div class="alert alert-dark mx-5" role="alert">
+        <strong>No orders for this branch are present.</strong>
+      </div>
+    <% else %>
+      <table class="table table-striped">
```
Comment (Sparsh-Goyal-2000) [inline app/views/admin/branches/show.html.erb:12]: We should cart orders to admin.
pls discuss with @himanshumishra31 sir, whether we should add filters here or not?

File: app/views/admin/branches/_order.html.erb [hunk 8]
```diff
@@ -0,0 +1,31 @@
+<tr class="<%= order_tr_class(order) %>">
+  <td><%= order.id %></td>
+  <td><%= order.user.first_name %></td>
+  <td>
+    <table class="table">
+      <thead>
+        <tr>
+          <th>Name</th>
```
Comment (Sparsh-Goyal-2000) [inline app/views/admin/branches/_order.html.erb:8]: No need to show name, quantity text repeatedly

**Review by Sparsh-Goyal-2000 — CHANGES_REQUESTED** [review]
Merge base brancha and resolve conflicts

File: config/routes.rb (hunk 2 above)
Comment (rishabh-vinsol) [reply config/routes.rb:42]: Done

Comment (rishabh-vinsol) [reply config/routes.rb:42]: Done

File: app/controllers/admin/branches_controller.rb (hunk 3 above)
Comment (rishabh-vinsol) [reply app/controllers/admin/branches_controller.rb:44]: By mistake updated this. Fixed now

File: app/controllers/orders_controller.rb (hunk 4 above)
Comment (rishabh-vinsol) [reply app/controllers/orders_controller.rb:71]: Done

File: app/models/order.rb (hunk 5 above)
Comment (rishabh-vinsol) [reply app/models/order.rb:45]: Done

File: app/models/order.rb (hunk 6 above)
Comment (rishabh-vinsol) [reply app/models/order.rb:49]: Done

File: config/routes.rb (hunk 2 above)
Comment (rishabh-vinsol) [reply config/routes.rb:42]: Done

File: app/views/admin/branches/_order.html.erb (hunk 8 above)
Comment (rishabh-vinsol) [reply app/views/admin/branches/_order.html.erb:8]: Done

File: db/migrate/20230608133616_add_url_slug_to_branch.rb (hunk 1 above)
Comment (rishabh-vinsol) [reply db/migrate/20230608133616_add_url_slug_to_branch.rb:3]: Done

File: app/views/admin/branches/_order.html.erb (hunk 8 above)
Comment (rishabh-vinsol) [reply app/views/admin/branches/_order.html.erb:8]: Done

File: db/migrate/20230608133616_add_url_slug_to_branch.rb (hunk 1 above)
Comment (rishabh-vinsol) [reply db/migrate/20230608133616_add_url_slug_to_branch.rb:3]: Done

File: app/views/admin/branches/show.html.erb [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+            <th>Picked Up At</th>
+            <th>Status</th>
+            <th>Change Status</th>
+            <th>Action</th>
+          </tr>
+          <tr>
+            <th></th>
+            <th></th>
+            <th style="text-align: left">Name</th>
+            <th style="text-align: right">Quantity</th>
```
Comment (Sparsh-Goyal-2000) [inline app/views/admin/branches/show.html.erb:35]: No need to show name quantity even for once

**Review by Sparsh-Goyal-2000 — APPROVED** [review]

**Pull request merged**